frontend/test/metabase/lib/expressions/completer.unit.spec.js

```diff
@@ -28,4 +31,35 @@ describe("metabase/lib/expressions/completer", () => {
       expect(partialMatch(" ")).toEqual(null);
     });
   });
+
+  describe("enclosingFunction", () => {
+    it("should get the correct name", () => {
+      expect(enclosingFunction("isnull([ID")).toEqual("isnull");
+    });
+
+    it("should ignore completed function construct", () => {
+      expect(enclosingFunction("Upper([Name])")).toEqual(null);
+    });
+
+    it("should handle multiple arguments", () => {
+      expect(enclosingFunction("Concat(First,Middle,Last")).toEqual("Concat");
+    });
+
+    it("should handle nested function calls", () => {
+      expect(enclosingFunction("Concat(X,Lower(Y,Z")).toEqual("Lower");
+      expect(enclosingFunction("P() + Q(R,S(7),T(")).toEqual("T");
+      expect(enclosingFunction("P() + Q(R,S(7),T()")).toEqual("Q");
+    });
+
+    it("should ignore non-function calls", () => {
+      expect(enclosingFunction("1")).toEqual(null);
+      expect(enclosingFunction("2 + ")).toEqual(null);
```

Comment:
Should this be operator, since space is handled a few lines below?
```suggestion
      expect(enclosingFunction("2 +")).toEqual(null);
```

Reply:
Good catch, thanks!